From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tony Hutter <hutter2@llnl.gov>
Date: Fri, 26 Oct 2018 01:24:24 -0400
Subject: [PATCH] Define timestruc_t for Lustre compatibility

Lustre 2.8 (and possibly other versions) are still using timestruc_t,
which was removed in spl-0.7.10 in favor of inode_timespec_t.  Add
in a backwards compatibility #define for timestruc_t so that Lustre
builds.

Reviewed by: Brian Behlendorf <behlendorf1@llnl.gov>
Reviewed-by: George Melikov <mail@gmelikov.ru>
Signed-off-by: Tony Hutter <hutter2@llnl.gov>
Closes #8014

Signed-off-by: Stoiko Ivanov <s.ivanov@proxmox.com>
---
 include/sys/time.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/sys/time.h b/include/sys/time.h
index 59557af..9029671 100644
--- a/include/sys/time.h
+++ b/include/sys/time.h
@@ -66,6 +66,9 @@ typedef struct timespec64	inode_timespec_t;
 typedef struct timespec		inode_timespec_t;
 #endif
 
+/* Include for Lustre compatibility */
+#define        timestruc_t     inode_timespec_t
+
 static inline void
 gethrestime(inode_timespec_t *ts)
  {
